posydon/binary_evol/DT/step_detached.py

```diff
@@ -936,79 +898,114 @@ def __repr__(self):
+            # we force primary.co=True for all isolated evolution,
+            # where the secondary is the one evolving one
             primary.co = True
-        elif (binary.star_1.state in ("BH", "NS", "WD")
-                and binary.star_2.state in LIST_ACCEPTABLE_STATES_FOR_HeStar):
-            primary = binary.star_1
+            primary.htrack = False
             secondary = binary.star_2
-            secondary.htrack = False
-            primary.htrack = secondary.htrack
-            primary.co = True
-        elif (binary.star_2.state in ("BH", "NS", "WD")
-                and binary.star_1.state in STAR_STATES_H_RICH):
-            primary = binary.star_2
-            secondary = binary.star_1
-            secondary.htrack = True
-            primary.htrack = secondary.htrack
-            primary.co = True
-        elif (binary.star_2.state in ("BH", "NS", "WD")
-                and binary.star_1.state in LIST_ACCEPTABLE_STATES_FOR_HeStar):
-            primary = binary.star_2
-            secondary = binary.star_1
-            secondary.htrack = False
-            primary.htrack = secondary.htrack
+            if (binary.star_2.state in STAR_STATES_H_RICH):
+                secondary.htrack = True
+            elif (binary.star_2.state in LIST_ACCEPTABLE_STATES_FOR_HeStar):
+                secondary.htrack = False
+            else:
+                raise Exception("State not recognized!")
+
+        elif self.non_existent_companion == 2:
             primary.co = True
-        elif (binary.star_1.state in STAR_STATES_H_RICH
-                and binary.star_2.state in STAR_STATES_H_RICH):
-            primary = binary.star_1
-            secondary = binary.star_2
-            secondary.htrack = True
-            primary.htrack = True
-            primary.co = False
-        elif (binary.star_1.state in LIST_ACCEPTABLE_STATES_FOR_HeStar
-                and binary.star_2.state in STAR_STATES_H_RICH):
-            primary = binary.star_1
-            secondary = binary.star_2
-            secondary.htrack = True
             primary.htrack = False
-            primary.co = False
-        elif (binary.star_2.state in LIST_ACCEPTABLE_STATES_FOR_HeStar
-                and binary.star_1.state in STAR_STATES_H_RICH):
-            primary = binary.star_2
             secondary = binary.star_1
-            secondary.htrack = True
-            primary.htrack = False
-            primary.co = False
-        elif (binary.star_1.state in LIST_ACCEPTABLE_STATES_FOR_HeStar
-                and binary.star_2.state in LIST_ACCEPTABLE_STATES_FOR_HeStar):
-            primary = binary.star_1
-            secondary = binary.star_2
-            secondary.htrack = False
-            primary.htrack = False
-            primary.co = False
-        else:
-            raise Exception("States not recognized!")
-
-        if not primary.co:
-            m1, t1 = get_mist0(primary, primary.htrack)
-        m2, t2 = get_mist0(secondary, secondary.htrack)
```

Comment:
This is actually to update htrack if the binary goes to the 3rd alternative matching. 
For the function get_star_data below, the input htrack is not updated if we call the matching function inside it. 
There must be a better way to deal with the binaries where we change the htrack.